src/commands/org/logout.ts

```diff
@@ -0,0 +1,215 @@
+    const maxAliasLength = Math.max(
+      'Aliases'.length,
+      ...orgAuths.map((orgAuth) => (orgAuth.aliases ? orgAuth.aliases.join(',') : '').length)
+    );
+    const maxConfigLength = Math.max(
+      'Configs'.length,
+      ...orgAuths.map((orgAuth) => (orgAuth.configs ? orgAuth.configs.join(',') : '').length)
+    );
+    const maxTypeLength = Math.max(
+      'Type'.length,
+      ...orgAuths.map((orgAuth) => {
+        if (orgAuth.isScratchOrg) {
+          return 'Scratch'.length;
+        }
+        if (orgAuth.isDevHub) {
+          return 'DevHub'.length;
+        }
+        if (orgAuth.isSandbox) {
+          return 'Sandbox'.length;
+        }
+        return 0;
+      })
+    );
+    const choices = orgAuths
+      .map((orgAuth) => {
+        const aliasString = (orgAuth.aliases ? orgAuth.aliases.join(',') : '').padEnd(maxAliasLength, ' ');
+        const configString = (orgAuth.configs ? orgAuth.configs.join(',') : '').padEnd(maxConfigLength, ' ');
+        const typeString = chalk.dim(
+          (orgAuth.isScratchOrg ? 'Scratch' : orgAuth.isDevHub ? 'DevHub' : orgAuth.isSandbox ? 'Sandbox' : '').padEnd(
+            maxTypeLength,
+            ' '
+          )
+        );
+        // username - aliases - configs
+        const key = `${chalk.bold(
+          orgAuth.username.padEnd(maxUsernameLength)
+        )} | ${typeString} | ${aliasString} | ${chalk.yellowBright(configString)}`;
+        return { name: key, value: orgAuth, checked: all, short: `${os.EOL}${orgAuth.username}` };
+      })
+      .sort((a, b) => a.value.username.localeCompare(b.value.username));
+    const userHeader = `${'Username'.padEnd(maxUsernameLength, ' ')}`;
+    const aliasHeader = `${'Aliases'.padEnd(maxAliasLength, ' ')}`;
+    const configHeader = `${'Configs'.padEnd(maxConfigLength, ' ')}`;
+    const typeHeader = `${'Type'.padEnd(maxTypeLength, ' ')}`;
+    return [new Separator(`  ${userHeader} | ${typeHeader} | ${aliasHeader} | ${configHeader}`), ...choices];
+  }
+
+  public async run(): Promise<AuthLogoutResults> {
+    const { flags } = await this.parse(Logout);
+    this.flags = flags;
+    this.configAggregator = await ConfigAggregator.create();
+    const remover = await AuthRemover.create();
+    let orgAuths: OrgAuthorization[] = [];
+    const targetUsername =
+      this.flags['target-org'] ?? (this.configAggregator.getInfo(OrgConfigProperties.TARGET_ORG).value as string);
+
+    // if no-prompt, there must be a resolved target-org or --all
+    if (flags['no-prompt'] && !targetUsername && !flags.all) {
+      throw messages.createError('noOrgSpecifiedWithNoPrompt');
+    }
+
+    orgAuths = this.shouldFindAllAuths(targetUsername)
+      ? await AuthInfo.listAllAuthorizations()
+      : targetUsername
+      ? await AuthInfo.listAllAuthorizations(
+          (orgAuth) => orgAuth.username === targetUsername || !!orgAuth.aliases?.includes(targetUsername)
+        )
+      : [];
```

Comment:
nit: since `orgAuths` isn't constant, could this be done with if/else conditionals? I found this a bit hard to follow (and still not sure I got it right).

Reply:
I'll look into simplifying for readability, if I can figure out what it is doing ;)